recipes/ignition-tools/all/conanfile.py

```diff
@@ -27,79 +29,104 @@ def _minimum_cpp_standard(self):
+        if os.name == "posix":
+            os.chmod(filename, os.stat(filename).st_mode | 0o111)
+
     def package(self):
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+
+        cmake = CMake(self)
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+        # Generate ign.rb
+        ign_rb_content = load(self, os.path.join(self.source_folder, "src", "ign.in"))
+        ign_rb_content = ign_rb_content.replace("@CMAKE_INSTALL_PREFIX@", self.package_folder.replace("\\", "/"))
+        ign_rb_content = ign_rb_content.replace("@ENV_PATH_DELIMITER@", os.pathsep)
+        suffix = ".rb" if self.settings.os == "Windows" else ""
+        ign_rb_path = os.path.join(self.package_folder, "bin", f"ign{suffix}")
+        save(self, ign_rb_path, ign_rb_content)
+        self._chmod_plus_x(ign_rb_path)
+
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
         # Remove MS runtime files
         for dll_pattern_to_remove in ["concrt*.dll", "msvcp*.dll", "vcruntime*.dll"]:
-            tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), dll_pattern_to_remove)
+            rm(self, dll_pattern_to_remove, os.path.join(self.package_folder, "bin"), recursive=True)
 
     def package_info(self):
-        version_major = tools.Version(self.version).major
-        self.cpp_info.names["cmake_find_package"] = "ignition-tools{}".format(version_major)
-        self.cpp_info.names["cmake_find_package_multi"] = "ignition-tools{}".format(version_major)
-
-        self.cpp_info.components["libignition-tools"].libs = ["ignition-tools-backward"]
-        self.cpp_info.components["libignition-tools"].includedirs.append("include/ignition/tools{}".format(version_major))
-        self.cpp_info.components["libignition-tools"].names["cmake_find_package"] = "ignition-tools{}".format(version_major)
-        self.cpp_info.components["libignition-tools"].names["cmake_find_package_multi"] = "ignition-tools{}".format(version_major)
-        self.cpp_info.components["libignition-tools"].names["pkg_config"] = "ignition-tools{}".format(version_major)
+        self.cpp_info.includedirs = []
+
+        version_major = Version(self.version).major
+        pkg_name = f"ignition-tools{version_major}"
+        self.cpp_info.set_property("cmake_file_name", pkg_name)
+        self.cpp_info.set_property("cmake_target_name", f"{pkg_name}::{pkg_name}")
+
+        component = self.cpp_info.components["libignition-tools"]
+        component.includedirs = []
```

Comment:
PLease, revert it, the package backward library provided by ignition-tool is not the `backward-cpp` package.

Reply:
Are you sure? It even includes a `revision` file in the directory, which links to https://github.com/bombela/backward-cpp/tree/3bb9240cb15459768adb3e7d963a20e1523a6294, which is the v1.6 tag. The backward.hpp file is identical. Since the backward-cpp library is used as header-only here, I would only add `libs=False` to the `requires()` call.

IMO it is preferable to unvendor unmodified external dependencies if possible, since it provides the consumer the option to override the version and it avoids potential silent conflicts if the dependency gets added somewhere else in the graph as well.

Reply:
If it's identical, so is fine using the external dependency 👍